extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php

```diff
@@ -9,203 +12,306 @@ class SpecialDiscussionsLogController extends WikiaSpecialPageController {
-		$logger = Wikia\Logger\WikiaLogger::instance();
-		$query = $this->getQuery( $userId );
-		$client = new GuzzleHttp\Client();
+			]
+			);
+		}
+		return $displayedUserLogRecords;
+	}
+
+	private function aggregateLogByUserId( $userId ) {
+		$query = UserQuery::getQuery( $userId, self::REQUEST_SIZE );
+		return $this->aggregateSearchLogs( $query );
+	}
+
+	private function aggregateLogByIpAddress( $ipAddress ) {
+		$query = IpAddressQuery::getQuery( $ipAddress, self::REQUEST_SIZE );
+		return $this->aggregateSearchLogs( $query );
+	}
+
+	private function aggregateSearchLogs( $query ) {
 		$records = [];
 		$dayOffset = 0;
+		$ipHash = [ ];
+
 		while ( count( $records ) < self::PAGINATION_SIZE && $dayOffset < self::DAYS_RANGE ) {
 			$url = $this->constructKibanaUrl( $dayOffset++ );
-			try {
-				$response = $client->post(
-					$url, [
-						'body' => $query,
-					]
-				);
-			} catch ( \GuzzleHttp\Exception\RequestException $requestException ) {
-				$logger->error(
-					sprintf( 'Request to elasticsearch failed: %s', $requestException->getMessage() ),
-					[ 'exception' => $requestException ]
-				);
-				break;
-			}
-
-			if ( $response->getStatusCode() !== self::HTTP_STATUS_OK ) {
-				$logger->error(
-					sprintf( 'Elasticsearch request error; status code %d', $response->getStatusCode() )
-				);
+			$response = $this->performSearchRequest( $url, $query );
+			if ( !$response ) {
 				break;
 			}
 
-			$resultObject = json_decode( $response->getBody() );
-			$hits = $resultObject->hits->hits;
-
-			foreach ( $hits as $hit ) {
-				$record = $hit->_source;
-
-				$rawTimestamp = $record->{'mobile_app.event.timestamp'};
-				$userLogRecord = new UserLogRecord();
-				$userLogRecord->app =
-					$record->{'mobile_app.data.app_name'} . ' ' . $record->{'mobile_app.data.app_version'};
-				$userLogRecord->ip = $record->{'mobile_app.client_ip'};
-				$userLogRecord->language = $record->{'mobile_app.event.device_language'};
-				$userLogRecord->location =
-					$record->{'mobile_app.geo_ip.city'} . ', ' . $record->{'mobile_app.geo_ip.country_name'};
-				$userLogRecord->timestamp = date( DATE_RFC2822, $rawTimestamp / 1000 );
-				$userLogRecord->userAgent =
-					$record->{'mobile_app.data.platform'} . ' ' . $record->{'mobile_app.data.platform_version'};
-				$userLogRecord->userId = $userId;
-				$userLogRecord->userName = $userName;
-				$records[ $rawTimestamp ] = $userLogRecord;
-			}
+			$this->appendRecordsFromResponse( $records, $response, $ipHash );
 		}
 
 		krsort( $records, SORT_NUMERIC );
 		return $records;
 	}
 
-	private function getQuery( $userId ) {
-		$pagination_size = self::PAGINATION_SIZE;
-		return <<<JSON_BODY
-{
-	"query": {
-		"filtered": {
-			"query": {
-				"bool": {
-					"should": [{
-						"query_string": {
-							"query":"mobile_app.event.user_id:$userId"
-						}
-					}]
-				}
-			}
+	private function performSearchRequest( $url, $query ) {
+		$client = new \GuzzleHttp\Client();
```

Comment:
I tried!
1. I cannot get it to work here!
2. There will be much less control over error handling
And honestly, I prefer GuzzleHttp because it is very reliable.